utils/jenkins_api.py

```diff
@@ -60,3 +61,44 @@ def unassign_role_from_user(self, role, user):
         )
 
         res.raise_for_status()
+
+    def list_plugins(self):
+        url_params = \
+            "depth=1&xpath=/*/*/shortName|/*/*/version&wrapper=plugins"
+        url = "{}/pluginManager/api/json?{}".format(self.url, url_params)
+
+        res = requests.get(
+            url,
+            verify=self.ssl_verify,
+            auth=(self.user, self.password)
+        )
+
+        res.raise_for_status()
+        return res.json()['plugins']
+
+    def install_plugin(self, name):
+        self.should_restart = True
+        header = {"Content-Type": "text/xml"}
+        url = "{}/pluginManager/installNecessaryPlugins".format(self.url)
+        data = \
+            '<jenkins><install plugin="{}@current" /></jenkins>'.format(name)
+        res = requests.post(
+            url,
+            verify=self.ssl_verify,
+            data=data,
+            headers=header,
+            auth=(self.user, self.password)
+        )
+
+        res.raise_for_status()
+
+    def safe_restart(self, force_restart=False):
```

Comment:
can we add a logging.debug in this method?

Reply:
done